packages/@aws-cdk/aws-stepfunctions-tasks/lib/run-glue-job-task.ts

```diff
@@ -0,0 +1,116 @@
+import * as glue from '@aws-cdk/aws-glue';
+import * as iam from '@aws-cdk/aws-iam';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import { Duration } from '@aws-cdk/core';
+import { getResourceArn } from './resource-arn-suffix';
+
+/**
+ * Properties for RunGlueJobTask
+ */
+export interface RunGlueJobTaskProps {
+
+  /**
+   * The service integration pattern indicates different ways to start the Glue job.
+   *
+   * The valid value for Glue is either FIRE_AND_FORGET or SYNC.
+   *
+   * @default FIRE_AND_FORGET
+   */
+  readonly integrationPattern?: sfn.ServiceIntegrationPattern;
+
+  /**
+   * The ID of a previous JobRun to retry.
+   *
+   * @default - Creates new run
+   */
+  readonly jobRunId?: string;
+
+  /**
+   * The job arguments specifically for this run. For this job run, they replace the
+   * default arguments set in the job definition itself.
+   *
+   * @default - Only arguments consumed by Glue
+   */
+  readonly arguments?: { [key: string]: string };
+
+  /**
+   * The number of AWS Glue data processing units (DPUs) to allocate to this JobRun.
+   *
+   * @default - 10
+   */
+  readonly allocatedCapacity?: number;
+
+  /**
+   * The JobRun timeout in minutes. This is the maximum time that a job run can consume
+   * resources before it is terminated and enters TIMEOUT status.
+   *
+   * @default - 2,880 (48 hours)
+   */
+  readonly timeout?: Duration;
+
+  /**
+   * The name of the SecurityConfiguration structure to be used with this job run.
+   *
+   * @default - No configuration
+   */
+  readonly securityConfiguration?: string;
+
+  /**
+   * Specifies configuration properties of a job run notification.
+   *
+   * @default - No configuration
+   */
+  readonly notificationProperty?: glue.CfnJob.NotificationPropertyProperty;
```

Comment:
It looks like the only property [supported by NotificationProperty is NotifyDelayAfter](https://docs.aws.amazon.com/glue/latest/dg/aws-glue-api-jobs-job.html#aws-glue-api-jobs-job-NotificationProperty). Let's model that here instead as a `Duration` type. Avoid using `glue.CfnJob.*` here.